code/game/objects/objs.dm

```diff
@@ -48,6 +48,11 @@ var/global/list/reagents_to_log = list(FUEL, PLASMA, PACID, SACID, AMUTATIONTOXI
 	if(src in processing_objects)
 		processing_objects -= src
 
+	if(integratedpai)
+		integratedpai.forceMove(get_turf(src))
```

Comment:
I really disagree with implicitly letting them live.

IMO deletion should recursively delete the contents no questions asked, because deletion is so without context and should always be taken as "get rid of it and leave no trace"